src/coreclr/vm/virtualcallstub.h

```diff
@@ -1550,6 +1550,13 @@ class FastTable
     {
         return ::operator new(baseSize + (numCallStubs + CALL_STUB_FIRST_INDEX) * sizeof(size_t));
     }
+public:
+    static void operator delete(void* ptr, size_t size)
+    {
+        // We calculate the size dynamically, so the size value here is
+        // invalid. Fall back to the unsized delete operator.
+        ::operator delete(ptr);
+    }
```

Comment:
```suggestion
    static void operator delete(void* ptr)
    {
        ::operator delete(ptr);
    }
```

I just realized we shouldn't need the size version. The default will be called appropriately. I think the real issue here is simply the allocator wasn't associated with the type and ASAN got grumpy.

Reply:
I looked at the ASAN implementation and it looks like it explicitly compares the allocation and deallocation size. Maybe if we have the unsized delete it won't try to use the sized one? I'll validate on godbolt.